llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -824,6 +825,48 @@ void InterBlockScheduling::emitInterBlockTop(const BlockState &BS) const {
   }
 }
 
+void InterBlockScheduling::emitInterBlockTop(BlockState &BS) {
+  if (BS.Kind != BlockType::Epilogue) {
+    return;
+  }
+
+  MachineBasicBlock *BB = BS.TheBlock;
+  MachineBasicBlock *Loop = getLoopPredecessor(*BB);
+  assert(Loop);
+  const BlockState &LBS = getBlockState(Loop);
+
+  // Epilogues of pipelined loops should emit the bundles swp epilog.
+  // in a dedicated exit. If there isn't one, spawn a new block,
+  // and a new block state and put this block to be scheduled later.
+  // Some maintenance of the original block state is also necessary.
+  if (LBS.isPipelined()) {
+    auto *DedicatedExit = makeDedicatedLoopExit(Loop, BB);
+    if (DedicatedExit == BB) {
+      // If we are in the same BB, just emit.
+      emitBundles(BS.TopInsert, DedicatedExit, DedicatedExit->begin(),
+                  /*Move=*/false, /*EmitNops=*/false);
+    } else {
+      // If not, create a new block state and transfer the timed region,
+      // Change old BS to regular, without TopInsert.
+      auto Itr = Blocks.emplace(DedicatedExit, DedicatedExit).first;
+      BlockState &NewBS = Itr->second;
+      NewBS.Kind = BlockType::Epilogue;
+      NewBS.TopInsert = BS.TopInsert;
+      BS.TopInsert.clear();
+      // We assume that BS will be a regular block now.
+      BS.Kind = BlockType::Regular;
+      // But BS can still be the epilogue of a DCL, let's
+      // check it.
+      llvm::for_each(BB->predecessors(), [L = BB, this](auto *S) {
+        if (getBlockState(S).Kind == BlockType::Loop) {
+          getBlockState(L).Kind = BlockType::Epilogue;
+        }
+      });
```

Comment:
as discussed, perhaps something along the lines `if (! any_of(predecessors, IsLoop))  BS.Kind =BLockType::Regular)`